_source/_posts/2022-06-14-simple-crud-react-and-spring-boot.md

```diff
@@ -0,0 +1,1348 @@
+        log.info("Request to create group: {}", group);
+        Group result = groupRepository.save(group);
+        return ResponseEntity.created(new URI("/api/group/" + result.getId()))
+                .body(result);
+    }
+
+    @PutMapping("/group/{id}")
+    ResponseEntity<Group> updateGroup(@Valid @RequestBody Group group) {
+        log.info("Request to update group: {}", group);
+        Group result = groupRepository.save(group);
+        return ResponseEntity.ok().body(result);
+    }
+
+    @DeleteMapping("/group/{id}")
+    public ResponseEntity<?> deleteGroup(@PathVariable Long id) {
+        log.info("Request to delete group: {}", id);
+        groupRepository.deleteById(id);
+        return ResponseEntity.ok().build();
+    }
+}
+```
+
+Add the following dependency to your `pom.xml` to fix compilation errors:
+
+```xml
+<dependency>
+  <groupId>jakarta.validation</groupId>
+  <artifactId>jakarta.validation-api</artifactId>
+  <version>2.0.2</version>
+</dependency>
+```
+
+If you stop and start your server app and hit `http://localhost:8080/api/groups` with your browser, or a command line client, you should see the list of groups.
+
+You can create, read, update, and delete groups with the following [HTTPie](https://httpie.org) commands.
+
+```bash
+http POST :8080/api/group name='Utah JUG' city='Salt Lake City' country=USA
+http :8080/api/group/6
+http PUT :8080/api/group/6 id=6 name='Utah JUG' address='On the slopes'
+http DELETE :8080/api/group/6
+```
+
+<!--
+<build>
+  <defaultGoal>spring-boot:run</defaultGoal>
+  ...
+</build>
+-->
+
+## Create a React UI with Create React App
+
+Create React App is a command line utility that generates React projects for you. It's a convenient tool because it also offers commands that will build and optimize your project for production. It uses webpack under the covers to build everything.
+
+Create a new project in the `jugtours` directory with `npx`.
+
+```bash
+npx create-react-app@5 app
+```
+
+After the app creation process completes, navigate into the `app` directory and install [Bootstrap](https://getbootstrap.com/), cookie support for React, React Router, and [Reactstrap](https://reactstrap.github.io/).
+
+```bash
+cd app
+npm i bootstrap@5 react-cookie@4 react-router-dom@6 reactstrap@9
+```
+<!-- npm install bootstrap@5.2.0-beta1 fixes autprefixer issue -->
+
+You'll use Bootstrap's CSS and Reactstrap's components to make the UI look better, especially on mobile phones. If you'd like to learn more about Reactstrap, see [reactstrap.github.io](https://reactstrap.github.io). It has extensive documentation on its various components and how to use them.
+
+Add Bootstrap's CSS file as an import in `app/src/index.js`.
+
+```js
+import 'bootstrap/dist/css/bootstrap.min.css';
+```
+
+## Call your Spring Boot API and display the results
+
+Modify `app/src/App.js` to use the following code that calls `/api/groups` and display the list in the UI.
```

Comment:
I see that you are not using any state management libarary like redux or mobx, for real world use cases they are nest parctice as `useState` still maintains state at component level which is not scalable, so may be a note about considering redux or mobx like library would be nice

Reply:
I haven't really needed redux or mobx for any of my simple React samples. Since I haven't used them, I don't think it's a good idea that I recommend them.